### PR TITLE
installer: refactor

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Borrow the harmless refactoring in https://github.com/NixOS/nixpkgs/pull/42610

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by samueldr** [comment]
Hi! Thanks for the PR!

Could you write a small human-friendly overview of the borrowed refactor?

(At a glance, it looks like it de-duplicates files)

File: nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  powerManagement.enable = true;
+
+  environment.systemPackages = [
+    # Include gparted for partitioning disks.
+    pkgs.gparted
+
+    # Include some editors.
+    pkgs.vim
+    pkgs.bvi # binary editor
+    pkgs.joe
```
Comment (worldofpeace) [inline nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix]: Do we need `joe` and `bvi`?

**Comment by worldofpeace** [comment]
It adds kate to `environment.systemPackages` in `installation-cd-graphical-kde.nix` because that's an expected consistency to at least have a gui text editor.

We still default to using slim for a display manager, but it's disabled for `installation-cd-graphical-gnome.nix` because we default to gdm there. Prior to this it was enabling both gdm and slim which obviously didn't work.

There is a comment which I no longer know is true
> GDM doesn't start in virtual machines with ISO

Verification is needed if this is still a problem (git blame says 3 years ago).

The rest is just de-duplication by adding `installation-cd-graphical-base.nix` for the shared configs between gnome and kde.

File: nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix (hunk 1 above)
Comment (adisbladis) [reply nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix]: Not really, though `bvi` only adds 136K and `joe` clocks in at 2.3M, hardly anything to worry about  (both on x86_64).

**Comment by adisbladis** [comment]
I had issues running `gnome` in qemu because it requires accelerated graphics. Maybe GDM is the same?

**Comment by worldofpeace** [comment]
It could be so, though I've had both working regularly in qemu.
Though testing shouldn't be important because `installation-cd-graphical-gnome.nix` isn't used.

**Comment by oxij** [comment]
Am I missing something or 2430a13bcaf96a9516e9347a4b675bb712e863bf removed manual desktop icon from gnome installer? Why?

**Comment by samueldr** [comment]
The [previous discussion](https://github.com/NixOS/nixpkgs/pull/42610#discussion_r202271131) explains that it would be because Gnome doesn't have a desktop anymore. (I haven't verified.)

**Comment by oxij** [comment]
I see, still, the bottom of `*-kde.nix` looks pretty useful for all graphical installers, is it not?

**Comment by oxij** [comment]
I mean, `konsole` part isn't but the rest is. Like if we ever want to make an Xfce installer (which I would wholeheartedly vote for, as it is the most KISS of the big three DEs) or something, it would get those icons automagically.

**Comment by worldofpeace** [comment]
@samueldr Yes GNOME removed this feature.

@oxij Maybe `installation-cd-graphical-base.nix` should have a `mkDesktopItem` (or something) in that case.

**Pull request merged**